app/controllers/comments_controller.rb

```diff
@@ -6,10 +6,22 @@ def create
     @comment = Comment.new(comment_params)
     @comment.author = current_user
     @comment.post = @post
+
+    authorize! :create, @comment
+
     @comment.save
     redirect_to user_post_path @post.author, @post
   end
 
+  def destroy
+    @comment = Comment.includes(:author, :post).find(params[:id])
+    @post = @comment.post
+    authorize! :create, @comment
+
+    @comment.destroy
+    redirect_to user_post_path @post.author, @post
+  end
+
```

Comment:
Take a look at **line 19**.

You are checking authorization for `:create` and the action is `:destroy`.

I actually checked the CanCanCan docs to make sure the action aliases are different:
![image](https://user-images.githubusercontent.com/55507343/206744925-a489974d-c94b-45e6-a5af-29b849a2560d.png)